harness/determined/common/experimental/metrics.py

```diff
@@ -0,0 +1,90 @@
+import dataclasses
+import datetime
+from typing import Any, Dict, List, Optional, cast
+
+from determined.common import util
+from determined.common.api import bindings
+
+
+@dataclasses.dataclass
+class TrialMetrics:
+    """
+    Specifies a metric that the trial reported.
+
+    Attributes:
+        trial_id
+        trial_run_id
+        total_batches
+        end_time
+        metrics
+        batch_metrics
+    """
+
+    trial_id: int
+    trial_run_id: int
+    total_batches: int
+    end_time: datetime.datetime
+    metrics: Dict[str, Any]
+    group: str
+    batch_metrics: Optional[List[Dict[str, Any]]] = None
+
+    @classmethod
+    def _from_bindings(cls, metric_report: bindings.v1MetricsReport, group: str) -> "TrialMetrics":
+        key = "validation_metrics" if group == util._LEGACY_VALIDATION else "avg_metrics"
+        return cls(
+            trial_id=metric_report.trialId,
+            trial_run_id=metric_report.trialRunId,
+            total_batches=metric_report.totalBatches,
+            end_time=util.parse_protobuf_timestamp(metric_report.endTime),
+            metrics=metric_report.metrics[key],
+            batch_metrics=metric_report.metrics.get("batch_metrics", None),
+            group=metric_report.group,
+        )
+
+    @property
+    def steps_completed(self) -> int:
+        """@deprecated: Use total_batches instead."""
+        return self.total_batches
+
+    @steps_completed.setter
+    def steps_completed(self, value: int) -> None:
+        self.total_batches = value
+
+
+class TrainingMetrics(TrialMetrics):
```

Comment:
wait, why do we even need to subclass TrialMetrics?  Why can't we just use TrialMetrics everywhere?

Reply:
I would prefer that but I wasn't sure how significant the blast radius is for accounting for the `steps_completed` and `total_batches` change that they automatically handle in cases where the user is explicitly asking for training metrics for example. I can try it, my new `.get_metrics()` function simply returns `TrialMetrics` as it is.

Reply:
We need to un-deprecate it.  Is the blast radius any smaller if you un-deprecate first, then rebase this work on top of that?

Reply:
Are you suggesting we make a new ticket to undeprecate these fields first? If so, I'm not sure as it would be completely tangential work to what this ticket is. My hunch is that it wouldn't really matter what order it happens in (either before or after this change)

Reply:
Added a ticket for this work: https://hpe-aiatscale.atlassian.net/browse/MLG-886

Reply:
I added deprecation warnings about using `TrainingMetrics` and `ValidationMetrics` in favor of using `TrialMetrics`. The functions that call these also warn about deprecation.

Reply:
I read a bunch more of the history around this code, and I'm happy now, and I agree with your approach.